hollow/src/main/java/com/netflix/hollow/api/producer/ProducerOptionalBlobPartConfig.java

```diff
@@ -0,0 +1,169 @@
+/*
+ *  Copyright 2021 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.api.producer;
+
+import com.netflix.hollow.api.consumer.HollowConsumer;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+/**
+ * A blob may be configured to produce optional "parts".  Parts will contain the data one or more types.
+ * 
+ * A type may be assigned to one and only one section -- by default a type will be serialized to the main "part".
```

Comment:
Curious about your thoughts on configurable "dual write" for producer i.e. if writing in parts then also write a full snapshot - so not all consumers need to have upgraded for a producer to onboard parts

Reply:
I'm not opposed.  I think this can be accomplished in a subsequent iteration if we find a need.